gateway/buckets.go

```diff
@@ -254,16 +259,35 @@ func (g *Gateway) renderWWWBucket(c *gin.Context, key string) {
 	for _, item := range rep.Item.Items {
 		if item.Name == "index.html" {
 			c.Writer.WriteHeader(http.StatusOK)
-			c.Writer.Header().Set("Content-Type", "text/html")
-			if err := g.buckets.PullPath(ctx, buck.Key, item.Name, c.Writer); err != nil {
+			contentType, r, err := getContentTypeFromPullPath(ctx, g.buckets, buck.Key, item.Name)
+			if err != nil {
 				render404(c)
 			}
+			c.Writer.Header().Set("Content-Type", contentType)
+			io.Copy(c.Writer, r)
 			return
 		}
 	}
 	renderError(c, http.StatusNotFound, fmt.Errorf("an index.html file was not found in this bucket"))
 }
 
+func getContentTypeFromPullPath(ctx context.Context, client *bucketsclient.Client, buckKey, pth string) (string, io.Reader, error) {
+	pr, pw := io.Pipe()
+	go func() {
+		defer pw.Close()
+		if err := client.PullPath(ctx, buckKey, pth, pw); err != nil {
+			log.Errorf("pulling path: %s", err)
+		}
+	}()
+
+	contentType, r, err := detectReaderContentType(pr)
+	if err != nil {
+		return "", nil, fmt.Errorf("detecting content-type: %s", err)
+	}
+
+	return contentType, r, nil
+}
```

Comment:
The magic is concentrated only here.
We give `PullPath` a writer from a Pipe, use the reader to let `detectReaderContentType` figure out the content-type and return a new reader (multireader).

Reply:
I believe this is the correct solution. Adding more bloat to `PullPath` API doesn't sound correct to me since not necessarily all clients are interested in content-type detection.